### PR TITLE
Tiled gallery: Add image filters

### PR DESCRIPTION
**Pull request by sirreal**
**Wait until Jetpack 7.1 is near before merging**

Implements #28932

Add a set of image filters to tiled gallery based on #28932.

## To do
- [x] Implement better vertical toolbar like in #28932 
- [x] Restructure to fix the known issue with applying the filter to the selected icon.
  Disabled the filter on selected image
- [x] Finish toolbar styling: https://github.com/Automattic/wp-calypso/pull/30911#issuecomment-466004499
- [x] Sync icons: https://github.com/Automattic/wp-calypso/issues/28932#issuecomment-464798243

## Testing instructions

* Open the block and change the filters.
* Try using the keyboard for the filters.
* Are the filter icons good?
* View the filter on the frontend (requires a WordPress.com patch or Jetpack site testing — gutenpack-jn)

### Toolbar

screen shot 2019-02-22 at 13 10 54

### Selecting an item with a filter
filter-workaround

### Original
orig

### Gingham
gingham

### Clarendon
clarendon

### Black and white
bw

### Sepia
sep

### CODE REVIEW

**Comment by sirreal** [comment]
Unfinished, but FYI: @thomasguillot @simison

File: client/gutenberg/extensions/tiled-gallery/filter-toolbar.js [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+					onClose();
+				};
+				return (
+					<Toolbar
+						icon={
+							<SVG width="24" height="24" viewBox="0 0 24 24">
+								<Rect fill="#f00" x="0" y="0" width="24" height="24" />
+							</SVG>
+						}
+						label={ __( 'Change Text Alignment' ) }
```
Comment (simison) [inline client/gutenberg/extensions/tiled-gallery/filter-toolbar.js]: >Change Text Alignment

:duck:

Comment (sirreal) [reply client/gutenberg/extensions/tiled-gallery/filter-toolbar.js]: Whoops. This needs to be changed anyways, I doubt it will be a toolbar at all.

**Comment by sirreal** [comment]
I reworked this using the same toolbar/menu as https://github.com/Automattic/wp-calypso/pull/30911#issuecomment-465997441

I'll need to wrap up the item styling on both of these.

I've removed the filter on the selected item, which seems like a good solution.

**Comment by sirreal** [comment]
- Rebased
- Updated icons: https://github.com/Automattic/wp-calypso/issues/28932#issuecomment-464798243
- Improved toolbar (matches WordAds https://github.com/Automattic/wp-calypso/pull/30952)

This should be ready!

**Comment by sirreal** [comment]
I've labeled this "DO NOT MERGE".

We'll plan to land this around the Jetpack 7.1 release.

**Comment by scruffian** [comment]
This is so good. Thanks :)

**Comment by sirreal** [comment]
Thanks folks! I'll squash this now so it can easily be picked to 7.1, then we can merge around release.

**Comment by sirreal** [comment]
Rebased and squashed. @jeherve this can be picked to a `7.1` branch at any time 👍

**Comment by eliorivero** [comment]
@thomasguillot why are the icons needed? Instagram doesn't use icons on their filter, neither does VSCO for example. I don't think I've ever seen an editor that uses icons, they all simply use labels or a preview of the filter applied to a thumbnail.
The 1977 icon is particularly troublesome, it looks redundant. Others like Sepia are confusing, why a flower? IMHO they serve no particular purpose or function and could be suppressed. We would avoid loading all those icons in millions of editors around the world. It's also ecologically greener to remove them than to include them.

icon

**Comment by thomasguillot** [comment]
@eliorivero looking at the icons again can indeed be troublesome/redundant/confusing.

We should probably switch to generic icons (cc @sirreal):

https://material.io/tools/icons/static/icons/outline-filter_1-24px.svg
https://material.io/tools/icons/static/icons/outline-filter_2-24px.svg
https://material.io/tools/icons/static/icons/outline-filter_3-24px.svg
https://material.io/tools/icons/static/icons/outline-filter_4-24px.svg
https://material.io/tools/icons/static/icons/outline-filter_5-24px.svg
https://material.io/tools/icons/static/icons/outline-filter_none-24px.svg

That being said, we need the icons to keep consistency with Gutenberg's design patterns.

screenshot 2019-03-04 at 16 19 40

**Comment by eliorivero** [comment]
> That being said, we need the icons to keep consistency with Gutenberg's design patterns.

Ah I see, that's understandable. Thanks for replying 👍

**Comment by sirreal** [comment]
@thomasguillot Update filter icons. Now:

screen shot 2019-03-04 at 18 25 11

@jeherve We'll need to pick 384ce09c120027265271220a0c271de077f55f45 into the 7.1 branch

**Comment by sirreal** [comment]
@jeherve Another change for 7.1 branch: 427800b71165c88bb0737248631c3e5464032f27

Deselect the selected image when a filter is applied.

Via p1HpG7-6lY-p2 #comment-30815

**Comment by jeherve** [comment]
Thank you! Done in 556b30359028842b6802326dd64ba05726bbd90a

**Comment by sirreal** [comment]
This was reverted, instead it will be landed as part of #31045

**Pull request merged**